src/v2/Components/v2/ArtworkFilter/ArtworkFilters/SizeFilter.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react"
+import { Box, Checkbox, Flex, Sans, Toggle } from "@artsy/palette"
+import { useArtworkFilterContext } from "../ArtworkFilterContext"
+
+const sizeMap = [
+  { displayName: "Small (under 40cm)", name: "SMALL" },
+  { displayName: "Medium (40 – 100cm)", name: "MEDIUM" },
+  { displayName: "Large (over 100cm)", name: "LARGE" },
+]
+
+/**
+ * Note: This is a clone of the implementation at
+ * src/v2/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/SizeFilter.tsx
+ * adapted to work with the common ArtworkFilter.
+ */
+export const SizeFilter: React.FC = () => {
+  const filterContext = useArtworkFilterContext()
+
+  const toggleSelection = (selected, name) => {
+    let sizes = filterContext.currentlySelectedFilters().sizes.slice()
+    if (selected) {
+      sizes.push(name)
+    } else {
+      sizes = sizes.filter(item => item !== name)
+    }
+    filterContext.setFilter("sizes", sizes)
+  }
+
+  return (
+    <Toggle label="Size" expanded>
+      <Flex flexDirection="column" alignItems="left">
+        <Sans size="2" color="black60">
+          This is based on the artwork’s average dimension.
+        </Sans>
+        <Box mt={0.25}>
+          {sizeMap.map((checkbox, index) => {
+            const { name, displayName } = checkbox
+            const props = {
+              key: index,
+              onSelect: selected => {
+                toggleSelection(selected, name)
+              },
+              selected: filterContext
+                .currentlySelectedFilters()
```

Comment:
Not totally opposed to duplication but wondering -- in the other `SizeFilter` `useArtworkFilterContext` is passed in as a prop due to it being shared in a few places cross-context. It might be worthwhile to use that one and pass in this filter context, but update it to check if `currentlySelectedFilters` exists. If so, execute it, if not, fall back to the default `filters` lookup (as seen in the other cloned component).

Reply:
Similar to above comment, it turned out that the presence of `currentlySelectedFilters` alone was not enough, we need all the machinery underneath it as well.